code/model/editableformfields/EditableFileField.php

```diff
@@ -73,6 +93,31 @@ public function getFormField() {
 		return $field;
 	}
 
+	/**
+	 * Validate if the file is over a certain size
+	 *
+	 * @return boolean
+	 */
+	public function validateField($data, $form) {
+		if ($this->MaxFileSize) {
+			foreach ($data as $file) {
+				if (isset($file['size'])) {
+					if (($this->MaxFileSize * 1024 * 1024) < $file['size']) {
+						$form->addErrorMessage($this->Name,
```

Comment:
You can set the filesize in EditableFileField::getField and save having to duplicate all of this validation.

```php
$field->getValidator()->setAllowedMaxFileSize(array('*' => $max));
```